Salesforce/sfdc_upload_lambda.py

```diff
@@ -0,0 +1,108 @@
+import os
+import json
+import boto3
+import requests
+import dateutil.parser
+from datetime import datetime, timedelta, timezone
+import time
+
+# add the following timestamp parse rule to the Sumo source: yyyyMMddHHmmss.SSS
+
+globs = {
+	"access_token": "", #UPDATE ME
+	"sumo_collector_url": "", # UPDATE ME
+	"client_id": "", # UPDATE ME
+	"refresh_token": "", # UPDATE ME
+	"refresh_url": "https://test.salesforce.com/services/oauth2/token",
+	"ssm": None,
+	"sfdc_url": f"https://{sfdc_instance}.salesforce.com/services/data/v32.0/",
+	"sfdc_instance": "" # UPDATE ME
+}
+
+def getAuthHeader():
+	access_token = globs["access_token"]
+	return {'Authorization': f"Bearer {access_token}"}
+
+def httpGet(url):
+	access_token = globs["access_token"]
+	response = requests.get(url, headers=getAuthHeader())
+
+	if (response.status_code >= 400):
+		print("token expired; getting a new one")
+		refresh_response = requests.post(
+			globs["refresh_url"], 
+			data={"grant_type": "refresh_token", "refresh_token": globs["refresh_token"], "client_id": globs["client_id"]},
+			headers={"Content-Type": "application/x-www-form-urlencoded"})
+		globs["access_token"] = refresh_response.json()['access_token']
```

Comment:
Should back up refresh_token as well, in case that's  changed.

Reply:
Where do you persist the globs params?

Reply:
Nowhere; it doesn't need to be persisted. The user just has to update those values once when they set up the lambda in their account.

Reply:
if the refresh_token is not persistent what happens if it becomes invalid? Doesn't it then break the lambda? Don't expect users to have to come in and update these values manually even once in a while.

Reply:
The only thing that would break the refresh token would be the user intentionally invaliding it via an API request that revokes it. So the token becoming invalid is not something to worry about.